src/main/java/org/pkcs11/jacknji11/jna/JNA.java

```diff
@@ -52,48 +52,60 @@
             ? ULong.ULongSize.ULONG4 : ULong.ULongSize.ULONG8;
     }
 
+    private JNANativeI jnaNative = null;
+    
+    public JNA(){
+        this("cryptoki");
+    }
+    
+    public JNA(String customLibrary) {
+        jnaNative = (JNANativeI) com.sun.jna.Native.loadLibrary(customLibrary, JNANativeI.class);
+    }
+
     public long C_Initialize(CK_C_INITIALIZE_ARGS pInitArgs) {
-        return JNANative.C_Initialize(new JNA_CK_C_INITIALIZE_ARGS(pInitArgs));
+        if(pInitArgs.createMutex == null && pInitArgs.destroyMutex == null && pInitArgs.lockMutex == null && pInitArgs.unlockMutex == null)
```

Comment:
Yes exactly this is a solution I applied for the #5 . It was there and I forgot to remove it.
Without this fix (or your solution) my pkcs11 library crash.